### ISSUE TITLE
[Feature Request] System time and date fetching

### ISSUE THREAD

**Issue by RedDevilus** [PCSX2/pcsx2#4491]
**Description**

Detach OS time and date from PCSX2 internal BIOS time and date
**Reason**

Some games rely on specific dates to work properly in PCSX2. It will also prohibit messing with the OS time and forgetting about it. 
**Examples**

![image](https://user-images.githubusercontent.com/24227051/121756661-2e85ac80-cb1b-11eb-8779-d742607135e5.png)

**Comment by maxdlpee** [comment]
May I ask for examples of what games require specific dates to work properly? AFAIK, no games require that, so this would be pointless.

**Comment by MrCK1** [comment]
Metal Gear Solid 3 relies on timed events (which use the system clock), so this could be helpful for fast forwarding those.

**Comment by kenshen112** [comment]
Metal Gear solid for one thing

**Comment by Mrlinkwii** [comment]
This may  also help  with speed running /input recording  , some game use the RTC as a determining  many RNG factors ( games such as FFX )

**Comment by Kryomancer** [comment]
To add onto this, Ratchet & Clank games feature secret developer museums that are only accessible at a specific hour late at night. Being able to set the clock manually without having to mess with the OS clock would make accessing this content a lot more convenient.

**Comment by F0bes** [comment]
I've looked into this, WX doesn't have a proper date time selector afaict. Maybe when we move to Qt this could be easier to do.

**Comment by kage2051** [comment]
> This may also help with speed running /input recording , some game use the RTC as a determining many RNG factors ( games such as FFX )

I agree. Dolphin has this setting for toggling at any time, and when this setting is enabled, it always throws the same RNG output. A pretty clear example is Tornado Outbreak on the Wii, where the animation after the second company logo (before the BinkVideo screen) is different depending on the system's clock. If this RTC value is always constant, the aforementioned animated screen never changes between runs or playbacks.

**Comment by TheTechnician27** [comment]
@RedDevilus What do you think now that v1.7 has long-since moved to Qt?

**Comment by RedDevilus** [comment]
It still makes sense to have a decoupling option to have a different time and date to system one. As being less intrusive than changing your computer real clock can have some weird side-effects or even forgetting to reset it to the correct time.

**Comment by crying2theocean** [comment]
please add this!

**Comment by refractionpcsx2** [comment]
Please don't bump issues unless you are being constructive, thank you.

**Comment by TheTechnician27** [comment]
Hey, Red, I think I'm going to start working on this. :) I'm busy with schoolwork, so it could take a while, but this has been up for 2.5 years at this point. Still, I think it's really important for speedrunners, glitch-hunters, and those who play games that rely on timed in-game events.

I'd like to add a tri-state checkbox which allows for both a global and per-game setting. In essence:

- If you don't have a per-game setting enabled, PCSX2 will default to the global setting.
- If you don't have a global setting, PCSX2 will default to the system time.

The range of dates will be from 2000-01-01 at 00:00:00 to 2099-12-31 at 23:59:59. The range of times technically depends on your BIOS' timezone, but for now, that seems like a bit overkill, so this just standardizes it to use Japan's range of times (GMT +9). The time naturally rolls over, so there should be no issue with letting users set the start time to that maximum, but anything outside of that range will automatically: a) be rounded up/down to the appropriate minimum/maximum, or b) be explicitly rejected and the user told to enter a valid value.

I think the option should go in the 'Emulation' tab. If this were just a global setting, I would put it under 'BIOS', but per-game settings don't have that.
